File: app/src/main/java/quintype/com/templatecollectionwithrx/models/sections/SectionMeta.java
package quintype.com.templatecollectionwithrx.models.sections;

import com.google.gson.annotations.SerializedName;

import quintype.com.templatecollectionwithrx.models.collection.CollectionMeta;

public class SectionMeta {
    @SerializedName("id")
    private String id;
    @SerializedName("display-name")
    private String displayName;
    @SerializedName("name")
    private String name;
    @SerializedName("parent-id")
    private String parentId;
    @SerializedName("slug")
    private String slug;
    @SerializedName("collection")
    private CollectionMeta collectionMeta;

    public String id() {
        return id;
    }

    public String displayName() {
        return displayName;
    }

    public String name() {
        return name;
    }

    public String parentId() {
        return parentId;
    }

    public String slug() {
        return slug;
    }

    public CollectionMeta collectionMeta() {
        return collectionMeta;
    }

    public void collectionMeta(CollectionMeta collectionMeta) {
        this.collectionMeta = collectionMeta;
    }

    @Override
    public String toString() {
        return "SectionMeta{" +
                "id='" + id + '\'' +
                ", displayName='" + displayName + '\'' +
                ", name='" + name + '\'' +
                ", slug='" + slug + '\'' +
                '}';
    }
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/sections/SectionMeta.kt
package quintype.com.templatecollectionwithrx.models.sections

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import quintype.com.templatecollectionwithrx.models.collection.CollectionMeta

/**
 * Created TemplateCollectionWithRx by rakshith on 8/31/18.
 */

class SectionMeta() : Parcelable {
    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeParcelable(collectionMeta, flags)
        dest?.writeString(this.id)
        dest?.writeString(this.displayName)
        dest?.writeString(this.name)
        dest?.writeString(this.parentId)
        dest?.writeString(this.slug)
    }

    override fun describeContents(): Int {
        return 0
    }

    @SerializedName("id")
    val id: String? = null
    @SerializedName("display-name")
    val displayName: String? = null
    @SerializedName("name")
    val name: String? = null
    @SerializedName("parent-id")
    val parentId: String? = null
    @SerializedName("slug")
    val slug: String? = null
    @SerializedName("collection")
    var collectionMeta: CollectionMeta? = null

    constructor(parcel: Parcel) : this() {
        collectionMeta = parcel.readParcelable(CollectionMeta::class.java.classLoader)
    }

    override fun toString(): String {
        return "SectionMeta{" +
                "id='" + id + '\''.toString() +
                ", displayName='" + displayName + '\''.toString() +
                ", name='" + name + '\''.toString() +
                ", slug='" + slug + '\''.toString() +
                '}'.toString()
    }

    companion object CREATOR : Parcelable.Creator<SectionMeta> {
        override fun createFromParcel(parcel: Parcel): SectionMeta {
            return SectionMeta(parcel)
        }

        override fun newArray(size: Int): Array<SectionMeta?> {
            return arrayOfNulls(size)
        }
    }
}